src/Controller/EnMarche/SearchController.php

```diff
@@ -18,11 +19,24 @@ class SearchController extends Controller
 {
     /**
      * @Route("/evenements", name="app_search_events")
+     * @Route("/evenements/categorie/{slug}", name="app_search_events_by_category")
      * @Method("GET")
      */
-    public function searchEventsAction(Request $request)
+    public function searchEventsAction(Request $request, string $slug = null)
     {
-        $request->query->set(SearchParametersFilter::PARAMETER_TYPE, SearchParametersFilter::TYPE_EVENTS);
+        if ($slug) {
+            if ($category = $this->getDoctrine()->getRepository(EventCategory::class)->findOneBy(['slug' => $slug])) {
+                $request->query->set(SearchParametersFilter::PARAMETER_TYPE, SearchParametersFilter::TYPE_EVENTS);
+                $request->query->set(SearchParametersFilter::PARAMETER_EVENT_CATEGORY, $category->getId());
+            } else if ($category = $this->getDoctrine()->getRepository(CitizenActionCategory::class)->findOneBy(['slug' => $slug])) {
+                $request->query->set(SearchParametersFilter::PARAMETER_TYPE, SearchParametersFilter::TYPE_CITIZEN_ACTIONS);
+                $request->query->set(SearchParametersFilter::PARAMETER_EVENT_CATEGORY, SearchParametersFilter::TYPE_CITIZEN_ACTIONS);
+            } else {
```

Comment:
you don't need to check CP category?

Reply:
No, it's only for events and citizen actions.